src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs

```diff
@@ -44,7 +44,7 @@ public unsafe void ToManaged(out Task? value)
 
             GCHandle gcHandle = (GCHandle)slot.GCHandle;
             JSHostImplementation.TaskCallback? holder = (JSHostImplementation.TaskCallback?)gcHandle.Target;
-            if (holder == null) throw new NullReferenceException("JSHostImplementation.TaskCallback");
+            if (holder == null) throw new InvalidProgramException(SR.Format(SR.FailedToMarshalTaskCallback));
```

Comment:
`InvalidOperationException` instead of `InvalidProgramException` for the 3 cases in this file.

[InvalidProgramException is for](https://learn.microsoft.com/en-us/dotnet/api/system.invalidprogramexception?view=net-7.0):
`The exception that is thrown when a program contains invalid Microsoft intermediate language (MSIL) or metadata. Generally this indicates a bug in the compiler that generated the program. This exception is also thrown when internal runtime implementation limits have been exceeded by the program.
`

Reply:
thanks, I replaced all of them